python/ray/data/datasource/file_based_datasource.py

```diff
@@ -367,6 +379,20 @@ def write(
         # succeeds.
         return "ok"
 
+    def on_write_complete(
```

Comment:
let's only delete the directory created by ourselves, but not from users. Should we guard it with `try_create_dir = True`?